GTO docs (#199)
iterative/mlem.ai: content/docs/gto/command-reference/index.md

# Using GTO Commands

> GTO is a command line tool. Here, we provide the specifications, complete

Author: Thanks @francesco086. This is good point of view. I'm opening a thread based on your comment to keep a discussion re it in a single place.

Minor: If you check out DVC docs, you'll see there are docs for Studio and DVClive. We can put this "GTO documentation" there or here (I used mlem.ai cause it was easier for me). Or to a separate website, like `iterative.ai/doc` maybe? Not sure. We need some place to keep GTO docs anyway.

Major: explaining how to build a registry with DVC+GTO+MLEM. Good question where to put that. In this PR you can see I was going to put answers in /doc/gto/user-guide. I guess the Tutorial format would be the best for this, and we could add it to each product involved under Use Cases (e.g. [here](https://mlem-ai-requirement-bui-vdivzc.herokuapp.com/doc/use-cases/model-registry) it can be next or instead of "Pure MLEM Model registry"):
<img width="246" alt="image" src="https://user-images.githubusercontent.com/6797716/197825960-c378aca6-29d8-4ed8-a7b8-3adf9db07030.png">

The other option is to create a GS with this - but that would be way to heavy for Get Started. I guess Tutorial or blog post serves the purpose better.

Another place to have this is [Model Registry page in Studio docs](https://dvc.org/doc/studio/user-guide/model-registry/what-is-a-model-registry). But, not sure yet how UI (Studio) and CLI (GTO+DVC+MLEM Tutorial) could co-exist here. Maybe cross-links are a better approach than having this in Studio docs.

Again, good topic to think about 🤔 We also leave CML out of the picture above, it's also a possible part of a MR...

@tapadipti, have you had any discussion about setting up a DVC+GTO+MLEM Tutorial to complement Studio docs? Looks like it much needed, but I can't see we ever created something like that.

Reviewer: I think it's a good place to put GTO docs if we want docs beyond a CMD/API ref (otherwise we could do with a README and possibly a site like https://docs.iterative.ai/dvc-task/reference/dvc_task/

---

>  you check out DVC docs, you'll see there are docs for Studio and DVClive

I think we should follow something similar to https://dvc.org/doc/dvclive here.

> Major: explaining how to build a registry with DVC+GTO+MLEM. Good question where to put that...
Tutorial format would be the best for this

We mention it very high-level in https://mlem.ai/doc/use-cases/model-registry now. And there's the https://iterative.ai/model-registry solution separately. I guess I'm not sure how much we want to go into the details of this 3-way integration. May be a good blog topic indeed. Let's create a separate issue to discuss that, though?

Reviewer 2: https://iterative.ai/model-registry should have links to all relevant docs pages. But since the docs can't reside there, Studio docs look like the next best place to me for explaining how to build a registry with DVC+GTO+MLEM. We could create a `Use cases` section. But depending on how much and what content we need, a blog post may also suffice. And docs specific to the GTO cli should definitely be separate.

> If you check out DVC docs, you'll see there are docs for Studio and DVClive.

This is to be changed. We will host Studio docs separately in its own docs site (like CML) - although we don't have dates for this yet.

Author: Ok, so I'm trying to draft that blog post - please see https://www.notion.so/iterative/Tutorial-Model-Registry-in-Git-with-DVC-MLEM-and-GTO-af124368ce9f4523a568a7e1875c7af3 - high-level feedback would be appreciated.

Reviewer 2: Thanks @aguschin. I've left some comments in the draft blog post.

> # GTO Command Reference
> 
> GTO is a command line tool (see [installation](/doc/gto/install)). For a list of
> all commands, type `gto -h`.

Reviewer: Not 100% sure on this one, up to you:

```suggestion
# GTO Command Reference

GTO is a command line too. It works on top of Git. For a list of all commands,
type `gto -h`.
```

> # Using GTO Commands

Author: This is very early WIP. Many pages are just placeholders yet. Want to get feedback about the docs structure, suggested pages (even if they're empty - learn if page worth to be created), and Get Started page. cc @jorgeorpinel as always, and @francesco086 - you've been contributing a lot recently, and you're using GTO, so your feedback would be valuable here.

Reviewer: > Want to get feedback about ... the docs structure

I don't think we need sections for use cases (there is one) or a user guide even, check https://dvc.org/doc/dvclive or even https://cml.dev/doc for example.

Focusing on Get Started for now.

Reviewer: p.s. I think we should remove many TODO work from this PR so it's easier to review and faster to merge.

Reviewer: Also, let's try to keep page titles consistent with URLs and nav entries:

```suggestion
# GTO Command Reference
```

> I just noticed this is an issue inherited from MLEM and [DVC docs](https://dvc.org/doc/command-reference)...

# Using GTO Commands

> GTO is a command line tool. Here, we provide the specifications, complete
> descriptions, and comprehensive usage examples for different `gto` commands.
> 
> For a list of all commands, type `gto -h`.

Reviewer: Minor, but how about we summarize a little? I'm not sure we'll end up providing all those things anyway:

```suggestion
GTO is a command line tool. For a list of all commands, type `gto -h`.
```

## Read the registry

- Show the registry state, highest version, or what's assigned in stage with
  `gto show`

> - Find out the artifact version registered/assigned with ref with

Reviewer: Hmmm... No need to have a full list of all commands in the index I think? This is more elaborate than what's in https://mlem.ai/doc/command-reference, for example. I think it's better to stick to the **Typical GTO workflow** commands here as well, to let users know which ones are "more important" (esp. since the nav is in alphabetical order).

Reviewer: Maybe something like

## Typical GTO workflow

- Make a registry with x and y commands
- Read the registry with ...
- Modify artifacts with ...
- Manage artifact enrichments with ...

Reviewer: p.s. unclear if you just land here what the difference between "Modify artifacts" and "Manage artifact enrichments" is though...
